boost-maven/boost-maven-plugin/src/main/java/io/openliberty/boost/utils/MavenProjectUtil.java

```diff
@@ -37,5 +39,25 @@ public static String findSpringBootVersion(MavenProject project) {
     public static boolean isNotNullOrEmpty(String s) {
         return s != null && !s.isEmpty();
     }
+    
+    /**
+     * Get all dependencies with "spring-boot-starter-*" as the artifactId. These
+     * dependencies will be used to determine which additional Liberty features need
+     * to be enabled.
+     * 
+     */
+    public static List<String> getSpringBootStarters(MavenProject project) {
```

Comment:
Instead of getting the starters, I think we should look at the complete list of dependencies (including transitive) and check for the specific underlying Spring Framework dependencies. For example, instead of checking for `spring-boot-starter-web` we should check for `org.springframework:spring-webmvc` which is the underlying Spring MVC implementation. That way it is more reliable if the user has a Spring Boot app that is using Spring Framework dependencies, for whatever reason.

Also, for reliability I think we should be checking both the group ID and artifact ID, not just the artifact ID.